### PR TITLE
Rework gbif cleanup

### PR DESCRIPTION
**Pull request by wendelfabianchinsamy**
## Purpose

closes: _Add github issue that originated this PR_

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)

- [ ] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Reviewer, please remember our [guidelines](https://datacite.atlassian.net/wiki/spaces/TEC/pages/1168375809/Pull+Request+Guidelines):

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Offer suggestions on how to improve code e.g. simplification or expanding clarity.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

File: app/jobs/delete_gbif_events_job.rb [hunk 1]
```diff
@@ -3,11 +3,18 @@
...
-  def perform(id, options = {})
-    event = Event.find_by(uuid: id)
+  def perform(ids, options = {})
+    label = options[:label]
+    index = options[:index]
 
-    event.destroy! if event.present?
+    # delete event records from mysql
+    sql = ActiveRecord::Base.sanitize_sql_array(["DELETE FROM events WHERE id IN (?)", ids])
+    ActiveRecord::Base.connection.execute(sql)
```
Comment (ashwinisukale) [inline app/jobs/delete_gbif_events_job.rb:12]: When we delete the events using raw query, we are skipping the [`after_commit on: [:destroy]` ](https://github.com/datacite/lupo/blob/master/app/models/concerns/indexable.rb#L53)callback.

Comment (wendelfabianchinsamy) [reply app/jobs/delete_gbif_events_job.rb:12]: @ashwinisukale yes this is by design as I will be using the bulk api to delete the documents in the events index.

File: app/models/event.rb [hunk 2]
```diff
@@ -880,6 +880,41 @@ def self.loop_through_events(options)
...
+    filter = options[:filter] || {}
+    label = options[:label] || ""
+    job_name = options[:job_name] || ""
+    query = options[:query].presence
+    delete_count = 0
+    max_delete_count = options[:max_delete_count]
+
+    response = Event.query(query, filter.merge(page: { size: 1, cursor: [] }))
+
+    if response.size.positive?
```
Comment (ashwinisukale) [inline app/models/event.rb:895]: we should verify `if response.present? && response.respond_to?(:size) && response.size.positive?` response present, incase of an error

File: app/models/event.rb [hunk 3]
```diff
@@ -880,6 +880,41 @@ def self.loop_through_events(options)
...
+      while response.size.positive? && delete_count < max_delete_count
+        response = Event.query(query, filter.merge(page: { size: size, cursor: cursor }))
+
+        break unless response.size.positive?
```
Comment (ashwinisukale) [inline app/models/event.rb:899]: we can make little readable by checking it like this `break if response.size.zero?`

**Review by ashwinisukale — APPROVED** [review]
looks good, there are some refactoring changes needed, but as you said we will be deleting this code so no need to worry on refactoring part.

Thanks,

**Pull request merged**